### PR TITLE
if chain options are omitted write cert and chain to the file specifi…

### PR DESCRIPTION
**Pull request by maettu**
…ed in certOutput. E.g. for nginx

as discussed last week.

There is one point I am unsure how much checking there should be. Atm nonsensical combinations of chainOutput and chainFormat are possible; no enforcement of either both options present or absent.

### CODE REVIEW

File: bin/acmefetch [hunk 1]
```diff
@@ -321,19 +323,33 @@ sub getCertificates {
...
+                my $certFile = $cert->{certOutput};
+                my $chainData = convertCert($acme->chain(),'DER',$cert->{chainFormat});
+                saveAway($certFile);
+                my $fh = IO::File->new( $certFile, "w" )
+                    || die "Write $certFile: $!";
+                print $fh $certData;
+                print $fh $chainData;
+                $fh->close();
+            }
             saveAway($cert->{keyOutput});
```
Comment (oetiker) [inline bin/acmefetch]: there is quite substantial code duplication ... how about just doing:

```perl
if ($cert->{chainOutput}){
￼   $fh->close();
   saveAway($cert->{chainOutput});
   $fh = IO::File->new( $cert->{chainOutput}, "w" )
￼                     || die "Write $cert->{chainOutput}: $!";
}
```
on line 330 of the original code and replace fh2 with fh

**Review by oetiker — CHANGES_REQUESTED** [review]

**Comment by maettu** [comment]
updated

File: bin/acmefetch [hunk 2]
```diff
@@ -327,13 +329,18 @@ sub getCertificates {
...
+            
+            # chain file wanted -> write separate chain file, otherwise append to cert file
+            if ($cert->{chainOutput}){
+                my $chainFile = $cert->{chainOutput};
+                $fh->close();
+                saveAway($chainFile);
+                $fh = IO::File->new( $chainFile, "w" )
+                    || die "Write $chainFile: $!";
+            }
             my $chainData = convertCert($acme->chain(),'DER',$cert->{chainFormat});
```
Comment (oetiker) [inline bin/acmefetch]: there should be 

```perl
} else {
    $cert->{chainFormat} = $cert->{certFormat}
}
```

**Review by oetiker — CHANGES_REQUESTED** [review]

File: bin/acmefetch (hunk 2 above)
Comment (oetiker) [reply bin/acmefetch]: oops that comment ended up one line to low

**Comment by oetiker** [comment]
thanks

**Pull request merged**